src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -33,13 +33,13 @@ internal sealed partial class ManagedWebSocket : WebSocket
         private static readonly UTF8Encoding s_textEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
 
         /// <summary>Valid states to be in when calling SendAsync.</summary>
-        private static readonly WebSocketState[] s_validSendStates = { WebSocketState.Open, WebSocketState.CloseReceived };
+        private static readonly int s_validSendStates = WebSocketStateHelper.ToFlags(WebSocketState.Open, WebSocketState.CloseReceived);
```

Comment:
I was looking at the BitArray, but it really looked like an overkill for this case (0 to 6 consecutive enum values). Too bad we can't make the enum to be flags, as it is a public API and it has explicit values assigned

If there is smth else you had in mind that I wasn't aware of pls let me know @stephentoub

Reply:
> Too bad we can't make the enum to be flags, as it is a public API and it has explicit values assigned

Do we need to use the public WebSocketState?

Reply:
Added a flags enum in https://github.com/dotnet/runtime/pull/107662/commits/bbda9af13c46b26fcb7ccc5d61187578ef2c16d2